### PR TITLE
refactor: remove modal recoil state and introduce redux

### LINKED ISSUE #635: refactor: remove modal recoil state and introduce redux
### QA:
All modals should be working exactly like the current state.

### PR DESCRIPTION
**Pull request by He1DAr**
- Introducing Redux
- Refactoring modal recoil state
closes #635

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]
Looks like a nice way to start this process. 👍

File: src/components/modals/add-network.tsx [hunk 1]
```diff
@@ -2,18 +2,20 @@ import React from 'react';
 import { Box, color, ControlledModal, Flex, IconButton, Stack } from '@stacks/ui';
 import { IconX } from '@tabler/icons';
 import { Link, Text, Title } from '@components/typography';
-import { useModal } from '@common/hooks/use-modal';
+import { closeModal, getOpenedModal } from '@common/state/slices/modal';
```
Comment (beguene) [inline src/components/modals/add-network.tsx:5]: @He1DAr I thought we would prefer organising folders by features from now on.

**Review by beguene — APPROVED** [review]

File: src/components/modals/add-network.tsx (hunk 1 above)
Comment (kyranjamie) [reply src/components/modals/add-network.tsx:5]: +1 on this

Wouldn't `common/state/modal/modal.slice.ts` be more organising by feature?

File: src/common/state/slices/modal.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    closeModal: state => {
+      state.openedModal = undefined;
+    },
+  },
+});
+
+export const { openModal, closeModal } = slice.actions;
+
+export const getOpenedModal = () => useAppSelector(state => state.modal.openedModal);
```
Comment (kyranjamie) [inline src/common/state/slices/modal.ts:28]: Should we figure out a naming convention for selectors? I've been using `select` Prefixed naming.

File: package.json [hunk 3]
```diff
@@ -30,6 +30,7 @@
     "@react-aria/interactions": "3.5.0",
     "@react-spectrum/utils": "3.6.1",
     "@react-stately/toggle": "3.2.2",
+    "@reduxjs/toolkit": "^1.7.2",
```
Comment (kyranjamie) [inline package.json:33]: Try and use the `-E` flag when installing packages, so it pins to an exact version. This was a recommendation of a security firm (we have a CI job to lint this)

**Review by kyranjamie — APPROVED** [review]
Looks good @He1DAr, added a few points to discuss.

Modal makes sense to start with, as this was already handled with state, though I think it's best practice that modals have corresponding routes, (rather than state)

File: src/components/modals/add-network.tsx (hunk 1 above)
Comment (He1DAr) [reply src/components/modals/add-network.tsx:5]: I was under the impression modal is a global state, but looking now we can def colocate it with modal 👍

**Comment by He1DAr** [comment]
> Looks good @He1DAr, added a few points to discuss.
> 
> Modal makes sense to start with, as this was already handled with state, though I think it's best practice that modals have corresponding routes, (rather than state)

I like the modal routes, what are your thoughts on keeping the same functionality with this refactor and addressing modal routes as a separate feature?

File: src/common/state/slices/modal.ts (hunk 2 above)
Comment (He1DAr) [reply src/common/state/slices/modal.ts:28]: I'm good with both, I used `get` as it's more framework-agnostic (not associated with reselect) but I think both work well.

**Comment by kyranjamie** [comment]
> I like the modal routes, what are your thoughts on keeping the same functionality with this refactor and addressing modal routes as a separate feature?

Agreed, this isn't necessary for now.

File: src/common/state/slices/modal.ts (hunk 2 above)
Comment (kyranjamie) [reply src/common/state/slices/modal.ts:28]: @beguene @fbwoolf for input

I don't really care either way, but let's just pick one.

Comment (fbwoolf) [reply src/common/state/slices/modal.ts:28]: I would just go with `select` if it is already implemented in the wallet changes. It makes sense to me as prefix.

Comment (beguene) [reply src/common/state/slices/modal.ts:28]: I prefer ```select```.

**Pull request merged**